### PR TITLE
fix(basicfilter): rework condition set form UI

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/syndesisio/syndesis/issues/6222

@gashcrumb this is as close as I could get while working with the existing markup. I couldn't figure out the auto form components to be able to modify the markup to make this layout easier/better. I'm also not sure where to inject the paragraph of description text that goes above the form fields.

In a perfect world, each of the `<fieldset>` elements would have a nested `<div>` that wraps the form groups so that I could give that a `grid` layout to place the the form groups inside of it.

A more flexible route would be to then wrap the 3 form group elements that create the condition in a `<div>`, then wrap the action/array controls in another element, that would allow me to use `flex` layout, which works in IE (since `grid` does not).

But what's there works, too, I just had to use `absolute` positioning for the delete action button, and %/relative spacing between the form groups and the delete action so it scales properly down to mobile.

Does this look ok, or can you (or point me in the right direction to) change the markup? And the same for injecting the description text before the form fields?

### CODE REVIEW

File: app/ui-react/syndesis/src/modules/integrations/components/editor/ruleFilter/WithRuleFilterForm.tsx [hunk 1]
```diff
@@ -103,13 +104,15 @@ export const WithRuleFilterForm: React.FunctionComponent<
       },
       arrayDefinitionOptions: {
         arrayControlAttributes: {
-          className: 'col-md-3 form-group',
+          className: 'form-group with-rule-filter-form__action',
         },
+        /*
```
Comment (gashcrumb) [inline app/ui-react/syndesis/src/modules/integrations/components/editor/ruleFilter/WithRuleFilterForm.tsx]: We can delete this block.

**Comment by gashcrumb** [comment]
So the markup for each element in the array is produced [here](https://github.com/syndesisio/syndesis/blob/master/app/ui-react/packages/auto-form/src/widgets/FormArrayComponent.tsx) which maps those classes you specify in the array type.  

There's also a storybook for auto-form ([example](https://syndesisio.github.io/syndesis-react/auto-form/index.html?path=/story/autoform--array)) so we could actually just take that form definition and plug it into that story or just create a new one which we could use to figure out the correct structure of the elements.  You'd just need to `cd` into `packages/auto-form` and then run `yarn storybook`.  

Now, the idea is that the form array stuff kinda needs to be really flexible so yeah, I kinda kept most of the layout logic open so that the code using auto-form kinda has to specify everything.

**Comment by gashcrumb** [comment]
> And the same for injecting the description text before the form fields?

This is a tough one, in my quick example I changed the `displayName` property and commented out the code that hides the label.  But then the problem with that is that text is then used in the validation prompt.  We *could* I guess add a new parameter but I dunno, that starts to feel a bit messy.

**Comment by mcoker** [comment]
@gashcrumb added an element inside the fieldset in the formarray component and converted the layout to grid. That works for me. Not sure about the description text though...

**Comment by gashcrumb** [comment]
Lemme see how it's looking here.

**Comment by gashcrumb** [comment]
So yeah, I think that's a huge improvement!  Looks like you've one lint error, but other than that LGTM

**Comment by gashcrumb** [comment]
As for the description text, one thought I had over the weekend at some point was I could probably just use a custom validator function for the basic filter page that just ensures validation fails but doesn't set the error text, which would then not hide the initial description text that's supposed to show up.  Let me look into that, I might push a commit with that onto this PR.

**Comment by mcoker** [comment]
@gashcrumb thanks!!

**Pull request merged**